crates/viewer/re_view_spatial/src/transform_cache.rs

```diff
@@ -0,0 +1,918 @@
+use std::collections::BTreeMap;
+
+use ahash::HashMap;
+use itertools::Either;
+use nohash_hasher::{IntMap, IntSet};
+
+use once_cell::sync::OnceCell;
+use re_chunk_store::{
+    ChunkStore, ChunkStoreSubscriberHandle, LatestAtQuery, PerStoreChunkSubscriber,
+};
+use re_entity_db::EntityDb;
+use re_log_types::{EntityPath, EntityPathHash, StoreId, TimeInt, Timeline};
+use re_types::{
+    components::{self},
+    Archetype as _, Component, ComponentName,
+};
+
+/// Store subscriber that resolves all transform components at a given entity to an affine transform.
```

Comment:
Does it keep track of the full transform tree, or just per-entity/instance transforms relative to their parents?
Who calculates the full transform tree (i.e. each `scene_from_entity`)?